### DISCUSSION TITLE
Best way to load a group of CGNS files

### DISCUSSION THREAD

**Discussion by eliesaikali** [CGNS/CGNS#747] [CGNStalk]
Dear all,

Due to some performance issues I am currently having by writing a single parallel cgns hdf file for my CFD simulation, I would like to test what gain I can have if I write a cgns file per processor. 

Working on a code that uses a domain splitting approach, I can manage to output the following 32 files (on 32 procs)

results_0000.cgns
results_0001.cgns
results_0002.cgns
...
results_0030.cgns
results_0031.cgns

Now, the question is the following : what is the best approach to read all these in paraview, or visit ?
Its clear that in paraview, we can load all these as a group. but in visit, this is not possible. I dont know how it goes with tecplot ...

Another question : is it possible to write a result.cgns file that points those 32 files so that I can just open results.cgns file in paraview/visit with no issues ?

Thanks for any help
Regards

**Comment by MicK7** [comment]
It is possible to create a single result.cgns file that points to multiple files through CGNS links (see cg_link_write API).
The limitation is on the number of simultaneous links opened.
